### PR TITLE
Add support for doing partial commit admin-level

### LINKED ISSUE #9: support for commit_changes_by?
Could the codebase be updated to support commit_changes_by for both FW and Panorama?

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/panos/_panos_commit.py#L190

https://github.com/PaloAltoNetworks/pango/blob/master/client.go#L416

https://github.com/PaloAltoNetworks/pango/blob/master/pano.go#L85

### PR DESCRIPTION
**Pull request by kairen**
Fixes #9.

### CODE REVIEW

**Comment by kairen** [comment]
\cc @shinmog

File: client.go [hunk 1]
```diff
@@ -1475,4 +1484,9 @@ type baseCommit struct {
 type baseCommitPartial struct {
     Dan string `xml:"device-and-network,omitempty"`
     Pao string `xml:"policy-and-objects,omitempty"`
+    Admin *baseCommitPartialAdmin `xml:"admin,omitempty"`
```
Comment (shinmog) [inline client.go]: If the admin listing is just a list of members, then use `util.MemberType` instead.  There is also the helpful `util.StrToMem()` that you can use.  So do this instead:

```golang
Admin *util.MemberType `xml:"admin"`
```

And finally, you can remove the `,omitempty` from this as pointers to structs are helpfully omitted if the pointers are `nil`.

File: client.go [hunk 2]
```diff
@@ -1475,4 +1484,9 @@ type baseCommit struct {
...
+}
+
+type baseCommitPartialAdmin struct {
```
Comment (shinmog) [inline client.go]: Can remove this when switching to `util.MemberType`.

File: client.go [hunk 3]
```diff
@@ -419,18 +422,24 @@ func (c *Client) UnlockCommits(vsys, admin string) error {
 // Commits result in a job being submitted to the backend.  The job ID and
 // if an error was encountered or not are returned from this function.  If
 // the job ID returned is 0, then no commit was needed.
-func (c *Client) Commit(desc string, dan, pao, force, sync bool) (uint, error) {
+func (c *Client) Commit(desc string, members []string, dan, pao, force, sync bool) (uint, error) {
```
Comment (shinmog) [inline client.go]: Can you change the variable name from "members" to "admins" to make it clearer what it's for?

File: client.go [hunk 4]
```diff
@@ -419,18 +422,24 @@ func (c *Client) UnlockCommits(vsys, admin string) error {
...
     c.LogAction("(commit) %q", desc)
 
     req := baseCommit{Description: desc}
-    if !dan || !pao {
+    if members != nil || !dan || !pao {
```
[the remark this answers is not in the document]
Comment (kairen) [reply client.go]: Yes, I agree with you! I will update it.

File: client.go (hunk 3 above)
Comment (kairen) [reply client.go]: Updated.

File: client.go (hunk 2 above)
Comment (kairen) [reply client.go]: Sure, I updated.

File: client.go (hunk 1 above)
Comment (kairen) [reply client.go]: Thanks for your feedback, this change is cleanly!

File: client.go [hunk 5]
```diff
@@ -419,18 +422,25 @@ func (c *Client) UnlockCommits(vsys, admin string) error {
...
-    if !dan || !pao {
+    if len(admins) > 0 || !dan || !pao {
         req.Partial = &baseCommitPartial{}
         if !dan {
             req.Partial.Dan = "excluded"
         }
         if !pao {
             req.Partial.Pao = "excluded"
         }
+	if len(admins) > 0 {
```
Comment (shinmog) [inline client.go]: You can just replace this whole code block (line 437 through 443) with this:

```golang
req.Partial.Admin = util.StrToMem(admins)
```

Comment (kairen) [reply client.go]: oh.. sorry! I didn't notice this util.. I have updated.

**Comment by gliptak** [comment]
Would https://github.com/PaloAltoNetworks/pango/blob/70f0705d29da85a039fcef44c3e6738f7eb8a265/pano.go#L88 need to be updated to make this available from Panorama?

**Pull request merged**